### PR TITLE
Add stubs for remaining commands in django.core.management.commands

### LINKED ISSUE #419: Missing most management commands
# Bug report

> Hi, thanks for submitting a bug. We appreciate that.

You're welcome.

> But, we will need some information about what's wrong to help you.

No problem.

## What's wrong

Stubs are missing for most of `django.core.management.commands`. Attempting to import one (to wrap in a custom command) results in the error `Module 'django.core.management.commands' has no attribute 'somecommand'`

## How is that should be

There should be stubs

## System information

- OS: macOS
- `python` version: 3.8.2
- `django` version: 2.2.14
- `mypy` version: 0.770
- `django-stubs` version: 1.5.0

### PR DESCRIPTION
**Pull request by timmartin**
This adds stubs for the remaining management commands in `django.core.management.commands`.

## Related issues

Closes #419

### CODE REVIEW

File: django-stubs/core/management/commands/compilemessages.pyi [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+def has_bom(fn: Any): ...
+def is_writable(path: Any): ...
+
+class Command(BaseCommand):
+    program: str = ...
+    program_options: List[str] = ...
+    verbosity: int = ...
+    has_errors: bool = ...
+    def compile_messages(self, locations: Any) -> None: ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/compilemessages.pyi]: Let's fix a bit of `Any` types! I am pretty sure that `locations` cannot be `Any`, can they?

File: django-stubs/core/management/commands/createcachetable.pyi [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+from django.core.cache import caches as caches
+from django.core.cache.backends.db import BaseDatabaseCache as BaseDatabaseCache
+from django.core.management.base import BaseCommand as BaseCommand, CommandError as CommandError
+from django.db import DEFAULT_DB_ALIAS as DEFAULT_DB_ALIAS, DatabaseError as DatabaseError, connections as connections, models as models, router as router, transaction as transaction
+from typing import Any
+
+class Command(BaseCommand):
+    verbosity: int = ...
+    def handle(self, *tablenames: Any, **options: Any) -> None: ...
+    def create_table(self, database: Any, tablename: Any, dry_run: Any) -> None: ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/createcachetable.pyi]: Looks like `dry_run` is `bool`, `database` is `str`, etc.

File: django-stubs/core/management/commands/dbshell.pyi [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from django.core.management.base import BaseCommand as BaseCommand, CommandError as CommandError, CommandParser as CommandParser
+from django.db import DEFAULT_DB_ALIAS as DEFAULT_DB_ALIAS, connections as connections
+from typing import Any
```
Comment (sobolevn) [inline django-stubs/core/management/commands/dbshell.pyi]: We don't need `Any` here

File: django-stubs/core/management/commands/flush.pyi [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from django.apps import apps as apps
+from django.core.management.base import BaseCommand as BaseCommand, CommandError as CommandError
+from django.core.management.color import Style, no_style as no_style
+from django.core.management.sql import emit_post_migrate_signal as emit_post_migrate_signal, sql_flush as sql_flush
+from django.db import DEFAULT_DB_ALIAS as DEFAULT_DB_ALIAS, connections as connections
+from typing import Any, Tuple
```
Comment (sobolevn) [inline django-stubs/core/management/commands/flush.pyi]: The same about `Any` here.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks!

Can you please:
1. Go through all new files and delete unused imports?
2. Remove as many `Any`s as possible

Thanks a lot for your contribution! 👍

File: django-stubs/core/management/commands/compilemessages.pyi [hunk 5]
```diff
@@ -1,13 +1,16 @@
+import os
 from django.core.management.base import BaseCommand as BaseCommand, CommandError as CommandError, CommandParser as CommandParser
 from django.core.management.utils import find_command as find_command, popen_wrapper as popen_wrapper
-from typing import Any, List
+from typing import List, Tuple, Union
 
-def has_bom(fn: Any): ...
-def is_writable(path: Any): ...
+PathType = Union[str, bytes, os.PathLike]
```
Comment (sobolevn) [inline django-stubs/core/management/commands/compilemessages.pyi]: We use `_PathType` notation when declaring our own type aliases.

File: django-stubs/core/management/commands/compilemessages.pyi [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
+
+def has_bom(fn: PathType): ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/compilemessages.pyi]: Should return `bool`

File: django-stubs/core/management/commands/compilemessages.pyi [hunk 7]
```diff
@@ -1,13 +1,16 @@
...
+def is_writable(path: PathType): ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/compilemessages.pyi]: The same

File: django-stubs/core/management/commands/squashmigrations.pyi [hunk 8]
```diff
@@ -7,9 +7,8 @@ from django.db.migrations.migration import SwappableTuple as SwappableTuple
...
 from django.db.migrations.writer import MigrationWriter as MigrationWriter
 from django.utils.version import get_docs_version as get_docs_version
-from typing import Any
 
 class Command(BaseCommand):
     verbosity: int = ...
-    interactive: Any = ...
-    def find_migration(self, loader: Any, app_label: Any, name: Any): ...
+    interactive: bool = ...
+    def find_migration(self, loader: MigrationLoader, app_label: str, name: str): ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/squashmigrations.pyi]: Missing return type

File: django-stubs/core/management/commands/showmigrations.pyi [hunk 9]
```diff
@@ -2,9 +2,9 @@ from django.apps import apps as apps
...
 from django.db.migrations.loader import MigrationLoader as MigrationLoader
-from typing import Any, Optional
+from typing import Any, List, Optional
 
 class Command(BaseCommand):
     verbosity: int = ...
-    def show_list(self, connection: Any, app_names: Optional[Any] = ...) -> None: ...
-    def show_plan(self, connection: Any, app_names: Optional[Any] = ...): ...
+    def show_list(self, connection: Any, app_names: Optional[List[str]] = ...) -> None: ...
+    def show_plan(self, connection: Any, app_names: Optional[List[str]] = ...): ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/showmigrations.pyi]: Missing return type

File: django-stubs/core/management/commands/migrate.pyi [hunk 10]
```diff
@@ -8,13 +8,13 @@ from django.db.migrations.loader import AmbiguityError as AmbiguityError
...
 class Command(BaseCommand):
     verbosity: int = ...
-    interactive: Any = ...
-    start: Any = ...
-    def migration_progress_callback(self, action: Any, migration: Optional[Any] = ..., fake: bool = ...) -> None: ...
-    def sync_apps(self, connection: Any, app_labels: Any): ...
+    interactive: bool = ...
+    start: float = ...
+    def migration_progress_callback(self, action: str, migration: Optional[Any] = ..., fake: bool = ...) -> None: ...
+    def sync_apps(self, connection: Any, app_labels: List[str]): ...
```
Comment (sobolevn) [inline django-stubs/core/management/commands/migrate.pyi]: Missing return type

**Review by sobolevn — CHANGES_REQUESTED** [review]
Awesome! Thanks!

**Comment by timmartin** [comment]
Thanks for the review. I believe I've addressed all your comments and picked up a couple more missing annotations as I went through. There are still some missing e.g. various DB commands take `connection` as an argument, which I think is an instance of `BaseDatabaseWrapper`, but it's hard to follow since we don't yet have a declaration for the connections in `django/db/__init__.py`.

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks for your help!
